sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/SqlKeywordsDelimitedBySpace.cs

```diff
@@ -0,0 +1,128 @@
+                "FROM DimEmployee AS e" + // Noncompliant
+                "ORDER BY LastName;"; // Noncompliant
+            var delete = "DELETE TOP 10 PERCENT" + "FROM target_table;"; // Noncompliant
+            var update = "UPDATE dbo.Table2" +
+                "SET dbo.Table2.ColB = dbo.Table2.ColB + dbo.Table1.ColB" + // Noncompliant
+                "FROM dbo.Table2"; // Noncompliant
+            var insert = "INSERT INTO CUSTOMERS (ID,NAME,AGE,ADDRESS,SALARY)" + // FN - we only consider alphanumeric characters
+                "VALUES(1, 'Ramesh', 32, 'Ahmedabad', 2000.00);";
+            var updateText = "UPDATETEXT @TableName.@ColumnName" +
+                "@ptrval @insert_offset"; // Noncompliant
+            var merge = "MERGE livesIn" +
+                "USING((SELECT @PersonId, @CityId, @StreetAddress) AS T(PersonId, CityId, StreetAddress)" + // Noncompliant
+                "    JOIN Person ON T.PersonId = Person.ID" +
+                "    JOIN City ON T.CityId = City.ID)" +
+                "  ON MATCH(Person-(livesIn)->City)" +
+                "WHEN MATCHED THEN" + // FN - we only consider alphanumeric characters
+                "  UPDATE SET StreetAddress = @StreetAddress" +
+                "WHEN NOT MATCHED THEN" + // Noncompliant
+                "  INSERT($from_id, $to_id, StreetAddress)" +
+                "  VALUES(Person.$node_id, City.$node_id, @StreetAddress);";
+            var writeText = "WRITETEXT pub_info.pr_info" + "ptrval 'text'"; // Noncompliant
+            var writeTextWithAt = "WRITETEXT pub_info.pr_info" + "@ptrval 'text'"; // Noncompliant
+            var readText = "READTEXT" + "pub_info.pr_info @ptrval 1 25;"; // Noncompliant
+        }
+
+        public string Property
+        {
+            get
+            {
+                return "SELECT something" + "FROM table"; // Noncompliant
+            }
+            set
+            {
+                value = "SELECT x" + "FROM table"; // Noncompliant
+            }
+        }
+
+        private string field = "SELECT *" +
+            "FROM TABLE" +  // Noncompliant
+            "WHERE X = 1;"; // Noncompliant
+
+        private string caseInsensitive = "select *" +
+            "from t" +  // Noncompliant
+            "where X = 1;"; // Noncompliant
+
+        private string onlyFirstKeyword = "select something" +
+            "because I want to";  // Noncompliant FP
+    }
+
+    class CompliantExamples
+    {
+        private string sql_string = "SELECT * FROM TABLE WHERE X = 1;";
+        private string sql_string_spaces_before = "SELECT *" +
+            " FROM TABLE" +
+            " WHERE X = 1;";
+        private string sql_string_spaces_after = "SELECT * " +
+            "FROM TABLE " +
+            "WHERE X = 1;";
+        private string not_sql_string = "I like tomatoes" + "and potatoes";
+        private int not_string = 1 + 2 + 3;
+        private string combined = "SELECT all" + 1 + "FROM table";
+
```

Comment:
I didn't find a test about `BinaryExpressionSyntax` that is not a `SyntaxKind.AddExpression`:
```cs
"SELECT " + 2 * 3 + "FROM table"
```
Or string concatenation containing parentheses, we should show one of those false-negative:
```cs
"SELECT " + ( "all" ) + "FROM table"
```